Make CUDA/ROCm/Ze plugins link against the existing ucs/ucm/uct libraries.
Rebased on UCX 1.22.0, where the plugin LDFLAGS gained $(UCX_LT_RELEASE).
UCX 1.22.0 links the CUDA perftest plugin against the in-tree libucxperf.la
whenever nvcc is found, so build that library from the plugin directory and
link it against the installed ucs/ucp/uct.

diff -ru a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -31,13 +31,13 @@
 AC_USE_SYSTEM_EXTENSIONS
 AC_CONFIG_HEADERS([config.h])
 
-AC_CHECK_PROG(GITBIN, git, yes)
-AS_IF([test x"${GITBIN}" = x"yes"],
-      [# remove preceding "refs/heads/" (11 characters) for symbolic ref
-       AC_SUBST(SCM_BRANCH,  esyscmd([sh -c 'git symbolic-ref --quiet   HEAD | sed "s/^.\{11\}//"']))
-       AC_SUBST(SCM_VERSION, esyscmd([sh -c 'git rev-parse    --short=7 HEAD']))],
-      [AC_SUBST(SCM_BRANCH,  "<unknown>")
-       AC_SUBST(SCM_VERSION, "0000000")])
+#AC_CHECK_PROG(GITBIN, git, yes)
+#AS_IF([test x"${GITBIN}" = x"yes"],
+#      [# remove preceding "refs/heads/" (11 characters) for symbolic ref
+#       AC_SUBST(SCM_BRANCH,  esyscmd([sh -c 'git symbolic-ref --quiet   HEAD | sed "s/^.\{11\}//"']))
+#       AC_SUBST(SCM_VERSION, esyscmd([sh -c 'git rev-parse    --short=7 HEAD']))],
+#      [AC_SUBST(SCM_BRANCH,  "<unknown>")
+#       AC_SUBST(SCM_VERSION, "0000000")])
 
 AH_TOP([
 #ifndef UCX_CONFIG_H
diff -ru a/src/ucm/cuda/Makefile.am b/src/ucm/cuda/Makefile.am
--- a/src/ucm/cuda/Makefile.am
+++ b/src/ucm/cuda/Makefile.am
@@ -9,7 +9,7 @@
 module_LTLIBRARIES      = libucm_cuda.la
 libucm_cuda_la_CPPFLAGS = $(BASE_CPPFLAGS) $(CUDA_CPPFLAGS)
 libucm_cuda_la_CFLAGS   = $(BASE_CFLAGS) $(CUDA_CFLAGS) $(LT_CFLAGS)
-libucm_cuda_la_LIBADD   = ../libucm.la $(CUDA_LIBS) $(CUDART_LIBS)
+libucm_cuda_la_LIBADD   = -lucm $(CUDA_LIBS) $(CUDART_LIBS)
 libucm_cuda_la_LDFLAGS  = $(UCM_MODULE_LDFLAGS) \
                           $(patsubst %, -Xlinker %, $(CUDA_LDFLAGS)) \
                           -version-info $(SOVERSION) $(UCX_LT_RELEASE)
diff -ru a/src/ucm/rocm/Makefile.am b/src/ucm/rocm/Makefile.am
--- a/src/ucm/rocm/Makefile.am
+++ b/src/ucm/rocm/Makefile.am
@@ -10,7 +10,7 @@
 module_LTLIBRARIES      = libucm_rocm.la
 libucm_rocm_la_CPPFLAGS = $(BASE_CPPFLAGS) $(ROCM_CPPFLAGS)
 libucm_rocm_la_CFLAGS   = $(BASE_CFLAGS) $(ROCM_CFLAGS) $(LT_CFLAGS)
-libucm_rocm_la_LIBADD   = ../libucm.la
+libucm_rocm_la_LIBADD   = -lucm
 libucm_rocm_la_LDFLAGS  = $(UCM_MODULE_LDFLAGS) \
                           $(ROCM_LDFLAGS) $(ROCM_LIBS) -version-info $(SOVERSION) \
                           $(UCX_LT_RELEASE) \
diff -ru a/src/uct/cuda/Makefile.am b/src/uct/cuda/Makefile.am
--- a/src/uct/cuda/Makefile.am
+++ b/src/uct/cuda/Makefile.am
@@ -12,8 +12,8 @@
 libuct_cuda_la_CFLAGS   = $(BASE_CFLAGS) $(CUDA_CFLAGS) $(LT_CFLAGS)
 libuct_cuda_la_LDFLAGS  = $(CUDA_LDFLAGS) -version-info $(SOVERSION) \
                           $(UCX_LT_RELEASE)
-libuct_cuda_la_LIBADD   = $(top_builddir)/src/ucs/libucs.la \
-                          $(top_builddir)/src/uct/libuct.la \
+libuct_cuda_la_LIBADD   = -lucs \
+                          -luct \
                           $(CUDA_LIBS) $(NVML_LIBS)
 libuct_cuda_ladir       = $(includedir)/uct/cuda
 
diff -ru a/src/uct/cuda/gdr_copy/Makefile.am b/src/uct/cuda/gdr_copy/Makefile.am
--- a/src/uct/cuda/gdr_copy/Makefile.am
+++ b/src/uct/cuda/gdr_copy/Makefile.am
@@ -11,7 +11,7 @@
 libuct_cuda_gdrcopy_la_LDFLAGS  = $(GDR_COPY_LDFLAGS) \
                                   -version-info $(SOVERSION) \
                                   $(UCX_LT_RELEASE)
-libuct_cuda_gdrcopy_la_LIBADD   = $(top_builddir)/src/ucs/libucs.la \
+libuct_cuda_gdrcopy_la_LIBADD   = -lucs \
                                   $(top_builddir)/src/uct/cuda/libuct_cuda.la \
                                   $(GDR_COPY_LIBS)
 
diff -ru a/src/uct/rocm/Makefile.am b/src/uct/rocm/Makefile.am
--- a/src/uct/rocm/Makefile.am
+++ b/src/uct/rocm/Makefile.am
@@ -8,8 +8,8 @@
 module_LTLIBRARIES      = libuct_rocm.la
 libuct_rocm_la_CPPFLAGS = $(BASE_CPPFLAGS) $(ROCM_CPPFLAGS)
 libuct_rocm_la_CFLAGS   = $(BASE_CFLAGS) $(LT_CFLAGS)
-libuct_rocm_la_LIBADD   = $(top_builddir)/src/ucs/libucs.la \
-                          $(top_builddir)/src/uct/libuct.la
+libuct_rocm_la_LIBADD   = -lucs \
+                          -luct
 libuct_rocm_la_LDFLAGS  = $(ROCM_LDFLAGS) $(ROCM_LIBS) -version-info $(SOVERSION) \
                           $(UCX_LT_RELEASE) \
                           $(patsubst %, -Xlinker %, -L$(ROCM_ROOT)/lib -rpath $(ROCM_ROOT)/hip/lib -rpath $(ROCM_ROOT)/lib) \
diff -ru a/src/uct/ze/Makefile.am b/src/uct/ze/Makefile.am
--- a/src/uct/ze/Makefile.am
+++ b/src/uct/ze/Makefile.am
@@ -10,8 +10,8 @@
 module_LTLIBRARIES    = libuct_ze.la
 libuct_ze_la_CPPFLAGS = $(BASE_CPPFLAGS) $(ZE_CPPFLAGS)
 libuct_ze_la_CFLAGS   = $(BASE_CFLAGS) $(LT_CFLAGS)
-libuct_ze_la_LIBADD   = $(top_builddir)/src/ucs/libucs.la \
-                          $(top_builddir)/src/uct/libuct.la
+libuct_ze_la_LIBADD   = -lucs \
+                        -luct
 libuct_ze_la_LDFLAGS  = $(ZE_LDFLAGS) $(ZE_LIBS) -version-info $(SOVERSION) \
                           $(UCX_LT_RELEASE) \
                           $(patsubst %, -Xlinker %, -L$(ZE_ROOT)/lib -rpath $(ZE_ROOT)/hip/lib -rpath $(ZE_ROOT)/lib) \
diff -ru a/src/tools/perf/cuda/Makefile.am b/src/tools/perf/cuda/Makefile.am
--- a/src/tools/perf/cuda/Makefile.am	2026-08-03 06:29:24.000000000 -0500
+++ b/src/tools/perf/cuda/Makefile.am	2026-10-30 10:04:24.879587614 -0500
@@ -30,6 +30,11 @@
     cuda_kernel.cuh
 
 include $(top_srcdir)/config/cuda.am
+
+# the plugin build only runs make in this directory, so build the in-tree
+# perf library it links here
+$(abs_top_builddir)/src/tools/perf/lib/libucxperf.la:
+	$(MAKE) -C $(abs_top_builddir)/src/tools/perf/lib libucxperf.la
 endif # HAVE_NVCC
 
 include $(top_srcdir)/config/module.am
diff -ru a/src/tools/perf/lib/Makefile.am b/src/tools/perf/lib/Makefile.am
--- a/src/tools/perf/lib/Makefile.am	2026-08-03 06:29:24.000000000 -0500
+++ b/src/tools/perf/lib/Makefile.am	2026-10-30 10:04:24.879526549 -0500
@@ -13,10 +13,7 @@
 libucxperf_la_CFLAGS   = $(BASE_CFLAGS) $(OPENMP_CFLAGS) $(LT_CFLAGS)
 libucxperf_la_CXXFLAGS = $(BASE_CXXFLAGS) $(OPENMP_CFLAGS)
 libucxperf_la_LDFLAGS  = $(RTE_LDFLAGS) $(OPENMP_CFLAGS)
-libucxperf_la_LIBADD   = \
-	$(abs_top_builddir)/src/uct/libuct.la \
-	$(abs_top_builddir)/src/ucp/libucp.la \
-	$(abs_top_builddir)/src/ucs/libucs.la
+libucxperf_la_LIBADD   = -luct -lucp -lucs
 
 # C-linkable C++ code - must override any inherited CXXFLAGS
 CXXFLAGS              += -nostdlib $(PERF_LIB_CXXFLAGS) -std=c++11
